### PR TITLE
feat: add href getter

### PR DESCRIPTION
**Pull request by anonrig**
Fixes https://github.com/ada-url/ada/issues/124

### CODE REVIEW

**Comment by lemire** [comment]
We also want origin getters !!! We either create a new issue, or else this PR gets updated.

File: include/ada/url.h [hunk 1]
```diff
@@ -25,41 +26,170 @@ namespace ada {
...
     std::string password{};
 
     /**
+     * @private
      * A URL’s host is null or a host. It is initially null.
      */
     std::optional<std::string> host{};
 
     /**
+     * @private
```
Comment (lemire) [inline include/ada/url.h:47]: Ah. Ah. I see you found those. 👍

**Comment by anonrig** [comment]
Let’s merge this and I can open another pull request for origin getter. @lemire

**Comment by lemire** [comment]
> By the way origin getter is in this pull request already.

Then it is great. That was my main concern.

**Comment by lemire** [comment]
@anonrig Consider extending the documentation in the README file.

**Comment by anonrig** [comment]
I'm done with this. Let's merge it @lemire

**Pull request merged**